src/auto-evo/MutationLogicFunctions.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace AutoEvo;
+
+using System.Linq;
+
+public class MutationLogicFunctions
```

Comment:
I think this + the mutation common functions would be best served to find a new home in a new folder like `auto-evo/mutations` and also bringing in the main Mutations.cs file there would be good for code grouping.